lib/deploy.ts

```diff
@@ -1,9 +1,102 @@
-import { BaseConfig } from "./config";
+import path from "path";
+import { exec, ExecException } from "child_process";
+import { BaseConfig, readConfig } from "./config";
+import { buildApp } from "./build";
+import { readWorkspace } from "./workspace";
+import { Network } from "./types";
+import { readdir, rename, remove } from "@/lib/utils/fs";
+
+const DEPLOY_DIST_FOLDER = "build";
+
+export type DeployOptions = {
+    deployAccountId: string;
+    signerAccountId: string;
+    signerPublicKey: string;
+    signerPrivateKey: string;
+    network?: Network;
+};
 
 // deploy the app widgets and modules
-export async function deployAppCode(src: string, config: BaseConfig) {
+export async function deployAppCode(src: string, dist: string, opts: DeployOptions) {
+    const deploying = log.loading(`[${src}] Deploying app`, LogLevels.BUILD);
+
+    const config = await readConfig(path.join(src, "bos.config.json"), opts.network);
+
+    // Move files from "src/widget" to "src/"
```

Comment:
Looks great, how about moving L25-53 to a "translate" or "translateForBosCli" function so we can test it?